src/response.js

```diff
@@ -8,61 +8,55 @@ export class DropboxResponse {
   }
 }
 
-export function parseResponse(res) {
+function throwIfError(res) {
   const clone = res.clone();
-
   if (!res.ok) {
-    return res.text()
+    return res.json()
       .then((data) => {
         // eslint-disable-next-line no-throw-literal
         throw {
           error: data,
           response: res,
           status: res.status,
         };
+      }, () => {
+        // eslint-disable-next-line no-throw-literal
+        throw {
+          error: clone.text(),
+          response: res,
+          status: res.status,
+        };
       });
   }
+  return Promise.resolve();
+}
+
+export function parseResponse(res) {
+  const clone = res.clone();
 
-  return res.json()
-    .catch(() => {
-      clone.text();
-    })
-    .then((data) => new DropboxResponse(res.status, res.headers, data));
+  return throwIfError(res).then(() => res.json()
```

Comment:
Having a promise return nothing (and used just for throwing errors) seems like an antipattern. Seems like we always do res.json() as a first step in all cases, so why not just do:

```
res.json().then((data) => {
  if (!res.ok) {
        throw {
          error: data,
          response: res,
          status: res.status,
        };
  }

  return new DropboxResponse(res.status, res.headers, data);
}).catch((err) => {
    throw {
      error: clone.text(),
      response: res,
      status: res.status,
    };
});
```

Reply:
In this case, that would work for the first function but for the second function, it wouldn't because we need to figure out if we should use `.json` `.blob` or `.binary`. I can do some further investigation to see if this can be simplified any more